gateway/api_definition.go

```diff
@@ -1728,13 +1728,20 @@ func (a *APISpec) getVersionFromRequest(r *http.Request) string {
 		// First non-empty part of the path is the version ID
 		for _, part := range strings.Split(uPath, "/") {
 			if part != "" {
-				if a.VersionDefinition.StripVersioningData || a.VersionDefinition.StripPath {
+				matchesUrlVersioningPattern := true
+				if a.VersionDefinition.UrlVersioningPattern != "" {
+					re := regexp.MustCompile(a.VersionDefinition.UrlVersioningPattern)
```

Comment:
Must panics, handle the error either:

- by printing a warning and ignoring the version regex as if its not there
- by printing a warning and resetting to a default version regex
- by printing an error and not loading the api (return the error)

We also have a /regexp package that caches the compilation, make sure its used here

Reply:
It's the best if we could do the regexp compilation while API is loaded, may in `EnabledForSpec`, so that it's only done once per reload.

Reply:
I'd rather prefer any of the above, as panics should not be error handling mechanisms for expected errors, but rather irrecoverable ones where no handling is possible. It's clearly possible to handle regex compilation errors without compromising the integrity of a running application with a panic.